analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocations.cs

```diff
@@ -136,42 +137,98 @@ End Sub
         }
 
         [DataTestMethod]
-        [DataRow("DoSomething();")]
-        [DataRow("this.DoSomething();")]
-        [DataRow("(this).DoSomething();")]
-        [DataRow("(((this))).DoSomething();")]
-        public void Invocation_InstanceMethodCallDoesNotClearFieldOnThis(string invocation)
+        [DataRow("Initialize();")]
+        [DataRow("this.Initialize();")]
+        [DataRow("(this).Initialize();")]
+        [DataRow("(((this))).Initialize();")]
+        [DataRow("((IDisposable)this).Dispose();")]
+        [DataRow("((IDisposable)(object)this).Dispose();")]
+        [DataRow("this.SomeExtensionOnSample();")]
+        [DataRow("Extensions.SomeExtensionOnSample(this);")]
+        [DataRow("this.SomeExtensionOnObject();")]
+        [DataRow("Extensions.SomeExtensionOnObject(this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((IDisposable)this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((object)(IDisposable)this);")]
+        [DataRow("((object)(IDisposable)this).SomeExtensionOnObject();")]
+        public void Invocation_InstanceMethodCallDoesClearFieldOnThis(string invocation)
```

Comment:
Add test cases for base calls: https://peach.sonarsource.com/project/issues?issues=AYMfm-_cBnZofnLJImz9&open=AYMfm-_cBnZofnLJImz9&id=aspnet-boilerplate

Reply:
There is at least one case in peach validation, where `this` is passed as a normal argument. Should we add support for this?

https://peach.sonarsource.com/code?id=dotnet-runtime&selected=dotnet-runtime%3Asrc/libraries/System.Data.Common/src/System/Data/DataTable.cs&line=275

Reply:
I would say it's not worth the trouble (yet).
In case `this` is passed somewhere, we should reset only public fields. At the same time, one should not use public fields. So the value is low.